src/v/cluster/tm_stm.cc

```diff
@@ -167,9 +167,13 @@ ss::future<checked<model::term_id, tm_stm::op_status>> tm_stm::do_barrier() {
         });
 }
 
+ss::future<> tm_stm::checkpoint_ongoing_txs() { return ss::now(); }
+
 ss::future<std::error_code>
 tm_stm::transfer_leadership(std::optional<model::node_id> target) {
-    return _c->do_transfer_leadership(target);
+    auto units = co_await _state_lock.hold_write_lock();
+    co_await checkpoint_ongoing_txs();
```

Comment:
I don't think this commit will compile / link since `checkpoint_ongoing_txs` isn't here. would it make sense to add the implementations _then_ wire it up?

Reply:
you're right. i didn't see it.